src/nspawn/nspawn-cgroup.c

```diff
@@ -135,21 +136,22 @@ int create_subcgroup(
 }
 
 int mount_cgroups(const char *dest, bool accept_existing) {
-        const char *p;
+        char *p;
+        _cleanup_close_ int fd = -EBADF;
         int r;
 
-        p = prefix_roota(dest, "/sys/fs/cgroup");
-
-        (void) mkdir_p(p, 0755);
+        r = chase("/sys/fs/cgroup", dest, CHASE_PREFIX_ROOT | CHASE_MKDIR_0755, &p, &fd);
+        if (r < 0)
+                return log_error_errno(r, "Failed to determine if a missing directory in %s was created: %m", p);
 
-        r = path_is_mount_point_full(p, dest, AT_SYMLINK_FOLLOW);
+        r = is_mount_point_at(fd, "", AT_SYMLINK_FOLLOW);
         if (r < 0)
                 return log_error_errno(r, "Failed to determine if %s is mounted already: %m", p);
         if (r > 0) {
                 if (!accept_existing)
                         return log_error_errno(SYNTHETIC_ERRNO(EEXIST), "Refusing existing cgroupfs mount: %s", p);
 
-                if (access(strjoina(p, "/cgroup.procs"), F_OK) >= 0)
+                if (faccessat(fd, strjoina(p, "/cgroup.procs"), F_OK, 0) >= 0)
```

Comment:
Then, please drop strjoina() and pass "cgroup.procs", otherwise fd will be anyway ignored.